app/src/main/java/de/grobox/transportr/trips/search/TripsFragment.kt

```diff
@@ -0,0 +1,175 @@
+/*
+ *    Transportr
+ *
+ *    Copyright (c) 2013 - 2018 Torsten Grote
+ *
+ *    This program is Free Software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as
+ *    published by the Free Software Foundation, either version 3 of the
+ *    License, or (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.grobox.transportr.trips.search
+
+import android.annotation.SuppressLint
+import android.content.Intent
+import android.os.Bundle
+import android.os.CountDownTimer
+import android.util.Pair
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Button
+import android.widget.Toast
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.omadahealth.github.swipyrefreshlayout.library.SwipyRefreshLayout.OnRefreshListener
+import com.omadahealth.github.swipyrefreshlayout.library.SwipyRefreshLayoutDirection
+import de.grobox.transportr.R
+import de.grobox.transportr.TransportrFragment
+import de.grobox.transportr.trips.detail.TripDetailActivity
+import de.grobox.transportr.trips.search.TripAdapter.OnTripClickListener
+import de.grobox.transportr.trips.search.TripsRepository.QueryMoreState
+import de.grobox.transportr.ui.LceAnimator
+import de.grobox.transportr.utils.Linkify
+import de.grobox.transportr.utils.TransportrUtils.getDragDistance
+import de.schildbach.pte.dto.Trip
+import kotlinx.android.synthetic.main.fragment_trips.*
+import java.util.regex.Pattern
+import javax.annotation.ParametersAreNonnullByDefault
+import javax.inject.Inject
+
+@ParametersAreNonnullByDefault
```

Comment:
I think this annotation isn't needed in Kotlin.

Reply:
Again, going to check. I'm not very familiar with these annotations.

Reply:
Removed too.